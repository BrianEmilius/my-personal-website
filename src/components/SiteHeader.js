import React from "react";
import Styled from "styled-components";
import { Link, StaticQuery, graphql } from "gatsby";

import PrimaryNavigation from "./PrimaryNavigation";

const Header = Styled.header`
	background-color: hsl(240, 30%, 20%);
  height: 5vh;
	display: flex;
	justify-content: space-between;
  line-height: 5vh;
  padding: 0 var(--pageMargin);
`;

const Image = Styled.img`
  height: 100%;
  width: auto;
`;

function Component({ data }) {
  return (
    <Header>
      <Link to="/">
        <Image src={data.file.publicURL} alt="icon in the shape of a house" />
      </Link>
      <PrimaryNavigation />
    </Header>
  );
}

const query = graphql`
  query {
    file(base: { eq: "home.svg" }) {
      publicURL
    }
  }
`;

export default function SiteHeader() {
  return (
    <StaticQuery query={query} render={data => <Component data={data} />} />
  );
}
