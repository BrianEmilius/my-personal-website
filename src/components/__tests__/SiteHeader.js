import React from "react";
import renderer from "react-test-renderer";
import { StaticQuery } from "gatsby";
import FileMock from "../../../__mocks__/file-mock";

import SiteHeader from "../SiteHeader";

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      home: {
        publicURL: FileMock
      },
      logo: {
        childImageSharp: {
          fluid: FileMock
        }
      }
    })
  );
});

describe("SiteHeader", function() {
  it("renders correctly", function() {
    const tree = renderer.create(<SiteHeader />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
