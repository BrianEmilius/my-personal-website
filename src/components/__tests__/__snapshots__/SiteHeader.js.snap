// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteHeader renders correctly 1`] = `
<header
  className="sc-htpNat eFupIx"
>
  <a
    href="/"
  >
    <img
      alt=""
      className="sc-bxivhb cmEYWm"
      src="test-file-stub"
    />
  </a>
  <nav>
    <ul
      className="sc-bdVaJa kfYGKV"
    >
      <li>
        <a
          className="sc-bwzfXH gNzaId"
          href="/blog"
        >
          Blog
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
